src/System.Text.Json/tests/Serialization/Value.WriteTests.GenericCollections.cs

```diff
@@ -285,6 +285,77 @@ public static void WritePrimitiveIReadOnlyListT()
             Assert.Equal("[1,2]", json);
         }
 
+        [Fact]
+        public static void WriteISetTOfISetT()
+        {
+            ISet<ISet<int>> input = new HashSet<ISet<int>>
+            {
+                new HashSet<int>() { 1, 2 },
+                new HashSet<int>() { 3, 4 }
+            };
+
+            Assert.Throws<NotSupportedException>(() => JsonSerializer.ToString(input));
+        }
+
+        [Fact]
+        public static void WriteISetTOfHashSetT()
+        {
+            ISet<HashSet<int>> input = new HashSet<HashSet<int>>
+            {
+                new HashSet<int>() { 1, 2 },
+                new HashSet<int>() { 3, 4 }
+            };
+
+            string json = JsonSerializer.ToString(input);
+            Assert.Equal("[[1,2],[3,4]]", json);
```

Comment:
`"[[3,4],[1,2]]"` is also valid (as is `"[[4,3],[1,2]]"`, etc); is there a way to structure these tests without the order dependence? Or is that okay given HashSet's implementation details are likely "fixed" (presuming folks depend on insert order being enumeration order).

Reply:
I've structured these tests without the order dependence - I think it's best.